src/daemon/ca/publishing.rs

```diff
@@ -1,232 +1,1223 @@
+            for event in events {
+                match event.details() {
+                    super::CaEvtDet::RoasUpdated {
+                        resource_class_name,
+                        updates,
+                    } => {
+                        objects.update_roas(resource_class_name, updates, timing, signer)?;
+                    }
+                    super::CaEvtDet::ChildCertificatesUpdated {
+                        resource_class_name,
+                        updates,
+                    } => {
+                        objects.update_certs(resource_class_name, updates, timing, signer)?;
+                    }
+                    super::CaEvtDet::KeyPendingToActive {
+                        resource_class_name,
+                        current_key,
+                    } => {
+                        objects.add_class(resource_class_name, current_key, timing, signer)?;
+                    }
+                    super::CaEvtDet::KeyPendingToNew {
+                        resource_class_name,
+                        new_key,
+                    } => {
+                        objects.keyroll_stage(resource_class_name, new_key, timing, signer)?;
+                    }
+                    super::CaEvtDet::KeyRollActivated {
+                        resource_class_name, ..
+                    } => {
+                        objects.keyroll_activate(resource_class_name, timing, signer)?;
+                    }
+                    super::CaEvtDet::KeyRollFinished { resource_class_name } => {
+                        objects.keyroll_finish(resource_class_name)?;
+                    }
+                    super::CaEvtDet::CertificateReceived {
+                        resource_class_name,
+                        rcvd_cert,
+                        ..
+                    } => {
+                        // Update the received certificate if needed. If the URIs changed we may need to re-issue things
+                        objects.update_received_cert(resource_class_name, rcvd_cert)?;
+                        objects.re_issue_if_required(&self.config.issuance_timing, &self.signer)?;
+                    }
+                    super::CaEvtDet::ResourceClassRemoved {
+                        resource_class_name, ..
+                    } => {
+                        objects.remove_class(resource_class_name);
+                    }
+                    super::CaEvtDet::RepoUpdated { contact } => {
+                        objects.update_repo(contact);
+                    }
+                    _ => {}
+                }
+            }
+            Ok(())
+        })
+    }
 }
 
-//------------ ManifestInfo ------------------------------------------------
+impl CaObjectsStore {
+    fn key(ca: &Handle) -> KeyStoreKey {
+        KeyStoreKey::simple(format!("{}.json", ca))
+    }
+
+    fn cas(&self) -> KrillResult<Vec<Handle>> {
+        let cas = self
+            .store
+            .read()
+            .unwrap()
```

Comment:
Is `unwrap()` the right thing to do here and not instead `read()?` or am I missing something?